Description: added the hardening CPPFLAGS and CXXFLAGS to configure.py
Author: Gary Kramlich <grim@reaperworld.com>
Index: ninja-1.3.3/configure.py
===================================================================
--- ninja-1.3.3.orig/configure.py	2013-05-24 21:38:46.000000000 -0500
+++ ninja-1.3.3/configure.py	2013-05-24 21:38:46.000000000 -0500
@@ -70,7 +70,7 @@ n.newline()
 
 n.comment('The arguments passed to configure.py, for rerunning it.')
 n.variable('configure_args', ' '.join(sys.argv[1:]))
-env_keys = set(['CXX', 'AR', 'CFLAGS', 'LDFLAGS'])
+env_keys = set(['CXX', 'AR', 'CFLAGS', 'LDFLAGS', 'CPPFLAGS', 'CXXFLAGS'])
 configure_env = dict((k, os.environ[k]) for k in os.environ if k in env_keys)
 if configure_env:
     config_str = ' '.join([k + '=' + configure_env[k] for k in configure_env])
@@ -176,6 +176,10 @@ def shell_escape(str):
 
 if 'CFLAGS' in configure_env:
     cflags.append(configure_env['CFLAGS'])
+if 'CPPFLAGS' in configure_env:
+    cflags.append(configure_env['CPPFLAGS'])
+if 'CXXFLAGS' in configure_env:
+    cflags.append(configure_env['CXXFLAGS'])
 n.variable('cflags', ' '.join(shell_escape(flag) for flag in cflags))
 if 'LDFLAGS' in configure_env:
     ldflags.append(configure_env['LDFLAGS'])
